src/picongpu/include/plugins/output/GatherSlice.hpp

```diff
@@ -55,7 +62,9 @@ struct GatherSlice
      */
     bool init(bool isActive)
     {
-        /*free old communicator of init is called again*/
+        static int masterRankOffset = 0;
```

Comment:
Side-note: The initialisation of function local statics is not guaranteed to be thread safe in pre-C++11 standards.